File: elixir/7ku/parts_of_a_list.ex
defmodule Partlist do
  def part_list(a) do
    Enum.map(1..(Enum.count(a) - 1), fn x ->
      [Enum.take(a, x) |> Enum.join(" "), Enum.take(a, x - Enum.count(a)) |> Enum.join(" ")]
    end)
  end
end

File: ruby/7ku/Parts of a list.rb
# frozen_string_literal: true

# Parts of a list
# https://www.codewars.com/kata/56f3a1e899b386da78000732/train/ruby

def partlist(a)
  (1...(a.length)).map { |x| [a[0, x].join(' '), a[x..-1].join(' ')] }
end

a = %w[az toto picaro zone kiwi]
p partlist(a)
